src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -34,8 +35,9 @@ public ODataSerializerContext()
         /// <param name="edmProperty">The complex property being nested or the navigation property being expanded.
         /// If the resource property is the dynamic complex, the resource property is null.
         /// </param>
+        /// <param name="expandedItems">The <see cref="ExpandedNavigationSelectItem"/> for the navigation property being expanded.</param>
         /// <remarks>This constructor is used to construct the serializer context for writing nested and expanded properties.</remarks>
-        public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty)
+        public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty, IList<ExpandedNavigationSelectItem> expandedItems = null)
```

Comment:
>ExpandedNavigationSelectItem [](start = 135, length = 28)

I want to implement $expand=property/navProperty/$ref, so, it's better to consider to use "ExpandedNavigationReferenceItem".